app/controllers/admin/cafe_works_controller.rb

```diff
@@ -0,0 +1,116 @@
+# encoding:UTF-8
+class Admin::CafeWorksController < ApplicationController
+  before_filter :authenticate
+  before_action :set_cafe_work, only: [:edit, :show, :update, :destroy, :remove_worker]
+
+  def show
+  end
+
+  def new
+    @cwork = CafeWork.new
+  end
+
+  def edit
+  end
+
+  def create
+    @cwork = CafeWork.new(c_w_params)
+    flash[:notice] = 'Cafejobbet skapades, success.' if @cwork.save
+    respond_with @cwork
+  end
+
+  def update
+    if @cwork.update(c_w_params)
+      flash[:notice] = 'Cafejobbet uppdaterades'
+      redirect_to @cwork
+    else
+      render action: :edit
+    end
+  end
+
+  def destroy
+    # id?
+    @id = @cwork.id
+    @cwork.destroy
+    respond_to do |format|
+      format.html { redirect_to :hilbert, notice: 'Cafepasset raderades.' }
+      format.js
+    end
+  end
+
+  def remove_worker
+    if !@cwork.clear_worker
+      render action: show, notice: "Lyckades inte"
+    end
+  end
+
+  def setup
+    @cwork = CafeWork.new
+  end
+
+  def setup_preview
```

Comment:
This means the method is too big and should be split. I do agree here, maybe you could make a helper method for setup_preview and setup_create? They are very similar.